core_types/src/link/mod.rs

```diff
@@ -63,6 +64,27 @@ pub enum LinkActionKind {
     REMOVE,
 }
 
+pub enum LinkMatch<S: Into<String>> {
+    Any,
+    Exactly(S),
+    Regex(S),
+}
+
+impl<S: Into<String>> LinkMatch<S> {
+    pub fn to_regex_string(self) -> Result<String, String> {
+        let re_string: String = match self {
+            LinkMatch::Any => ".*".into(),
+            LinkMatch::Exactly(s) => "^".to_owned() + &s.into() + "$",
+            LinkMatch::Regex(s) => s.into(),
+        };
+        // check that it is a valid regex
+        match Regex::new(&re_string) {
+            Ok(_) => Ok(re_string),
+            Err(_) => Err("Invalid regex passed to get_links".into()),
```

Comment:
Could be reduced to a one liner by using Regex::new(&re_string).map_err(|_|Err("Invalid regex passed to get_links".into())?;

Reply:
It actually must be 
```
Regex::new(&re_string)
    .map(|_| re_string)
    .map_err(|_| Err("Invalid regex passed to get_links".into() )?
```
I figured this is just as much code as the match and the match version is maybe slightly easier to read